www/src/components/publisher/CreatePublisherModal.tsx

```diff
@@ -0,0 +1,281 @@
+          placeholder={placeholder}
+        />
+      </FormField>
+    )
+  }
+
+  return (
+    <Modal
+      open={open}
+      onClose={onClose}
+      size="large"
+    >
+      <ModalHeader>
+        Create a publisher
+      </ModalHeader>
+      {iconPicker.HiddenFileInput(iconPickerInputOpts)}
+      <FormField
+        label="Icon"
+        marginBottom="large"
+      >
+        <Flex
+          direction="row"
+          alignItems="flex-end"
+          gap="medium"
+        >
+          <IconUploadPreview
+            src={iconUpload.previewUrl}
+            onClick={iconPicker.onClick}
+          />
+          <Flex
+            direction="column"
+            gap="xsmall"
+          >
+            <Button
+              type="button"
+              secondary
+              small
+              minHeight="auto"
+              onClick={iconPicker.onClick}
+            >
+              {iconUpload.previewUrl ? 'Switch' : 'Upload'}
+            </Button>
+            {iconUpload.previewUrl && (
+              <Button
+                type="button"
+                secondary
+                small
+                minHeight="auto"
+                destructive
+                onClick={() => {
+                  setIconUpload({ file: null, previewUrl: null })
+                }}
+              >
+                Delete
+              </Button>
+            )}
+          </Flex>
+        </Flex>
+      </FormField>
+      <FormField
+        label="Name"
+        marginBottom="large"
+      >
+        <Input
+          value={name}
+          onChange={event => setName(event.target.value)}
+          placeholder="Enter a name"
+        />
+      </FormField>
+      <FormField
+        label="Description"
+        marginBottom="large"
+        length={description.length}
+        maxLength={200}
+      >
+        <Input
+          multiline
+          minRows={3}
+          value={description}
+          onChange={event => setDescription(event.target.value.substring(0, 200))}
+          placeholder="Enter a description"
+        />
+      </FormField>
+      <Flex gap="medium">
+        {renderUrlField(
+          'Website link', 'Website URL', website, setWebsite, errors.website
+        )}
+        {renderUrlField(
+          'Docs link', 'Docs URL', documentation, setDocumentation, errors.documentation
+        )}
+      </Flex>
+      <Flex gap="medium">
+        {renderUrlField(
+          'GitHub link', 'GitHub URL', github, setGithub, errors.github
+        )}
+        {renderUrlField(
+          'Docs link', 'Discord invite URL', discord, setDiscord, errors.discord
+        )}
+      </Flex>
+      <Flex gap="medium">
+        {renderUrlField(
+          'Slack link', 'Slack invite URL', slack, setSlack, errors.slack
+        )}
+        {renderUrlField(
+          'Twitter link', 'Twitter URL', twitter, setTwitter, errors.twitter
+        )}
+      </Flex>
+      {!!preMutationError && (
+        <Alert severity="error">
```

Comment:
why don't we use the ValidatedInput component for these?  Showing full alerts will be a pretty jarring ux

Reply:
also it makes the code wayyy more verbose

Reply:
I'm pushing toward the alert because the error is usually that the owner already has a publisher created. It does not fit any of the fields.

Reply:
oh for errors from gql I think we should have an alert, for validation errors on the input fields themselves, something associated with the field makes more sense